core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -286,6 +286,34 @@ private[spark] class TaskSchedulerImpl(
     }
   }
 
+  /**
+   * SPARK-25250: Whenever any Result Task gets successfully completed, we simply mark the
+   * corresponding partition id as completed in all attempts for that particular stage and
+   * additionally, for a Result Stage, we also kill the remaining task attempts running on the
+   * same partition. As a result, we do not see any Killed tasks due to
+   * TaskCommitDenied Exceptions showing up in the UI.
+   */
+  override def completeTasks(partitionId: Int, stageId: Int, killTasks: Boolean): Unit = {
+    taskSetsByStageIdAndAttempt.getOrElse(stageId, Map()).values.foreach { tsm =>
+      tsm.partitionToIndex.get(partitionId) match {
+        case Some(index) =>
+          tsm.markPartitionCompletedForRedundantTaskAttempts(index)
```

Comment:
how about `markPartitionAsAlreadyCompleted`? This indicates that the partition is already completed by other task set managers, and this task set manager only needs to mark it as completed, but not to track its run time or increase the numSuccessfulTasks.

Reply:
Done.